modules/auxiliary/scanner/redis/redis_login.rb

```diff
@@ -39,7 +39,26 @@ def initialize(info = {})
     deregister_options('USERNAME', 'USER_AS_PASS', 'USERPASS_FILE', 'USER_FILE', 'DB_ALL_USERS', 'DB_ALL_CREDS', 'PASSWORD_SPRAY')
   end
 
+  def requires_password?(_ip)
+    connect
```

Comment:
I checked and it's not necessary to catch connection errors here because they just boil up to [`Msf::Auxiliary::Scanner`](https://github.com/rapid7/metasploit-framework/blob/48e120d4e959e60cb1f539185c23580a8cbc1f5f/lib/msf/core/auxiliary/scanner.rb#L128). The only disadvantage to letting it be handled at that level is that there's no error message, and bruteforcing won't be attempted (which will allow for a couple of connection errors if that phases is started).